### ISSUE TITLE
"use symbolic math" fontifies (incorrectly scaled) data structures

### ISSUE THREAD

**Issue by notuntoward** [spyder-ide/spyder#4556]
## Description
If I have checked

preferences --> ipython console --> advanced settings --> use symbolic math

Then data structures like dicts are fontified, which seems to be an unrelated behavior.  Also, the scaling is incorrect.  Here's a picture of the console output of dict, d:

![symbdisplay](https://cloud.githubusercontent.com/assets/3851891/26754853/abc00cba-4882-11e7-8cb0-71eb0257c99a.JPG)

**What is the expected output? What do you see instead?**

I would expect to see this:

![nosymbdisplay](https://cloud.githubusercontent.com/assets/3851891/26754858/b7c8cd9e-4882-11e7-822e-2f1a96ad81ec.JPG)

**Please provide any additional information below**

This is Windows 10, on a Surface Pro 4, and with 

preferences --> general --> appearance --> "enable high DPI scaling"

clicked. 

## Version and main components

* Spyder Version: 3.1.4 
* Python Version: 3.5.3
* Qt Versions: 5.6.2, PyQt5 5.6 on Windows

## Dependencies
```
pyflakes >=0.6.0 :  1.5.0 (OK)
pep8 >=0.6       :  1.7.0 (OK)
pygments >=2.0   :  2.2.0 (OK)
qtconsole >=4.2.0:  4.3.0 (OK)
nbconvert >=4.0  :  5.2.1 (OK)
pandas >=0.13.1  :  0.20.1 (OK)
numpy >=1.7      :  1.12.1 (OK)
sphinx >=0.6.6   :  1.5.6 (OK)
rope >=0.9.4     :  0.9.4-1 (OK)
jedi >=0.9.0     :  0.10.2 (OK)
matplotlib >=1.0 :  2.0.2 (OK)
sympy >=0.7.3    :  1.0 (OK)
pylint >=0.25    :  1.6.4 (OK)

```

**Comment by ccordoba12** [comment]
This is the expected result when you use symbolic math in the console. In the future we could provide more fine grained options to symbolic math, but for now there's nothing we can do about it, sorry.

**Comment by notuntoward** [comment]
There were two problems here.  The first was the one just closed; the second was that the font scaling was incorrect.  Can that be fixed?

**Comment by ccordoba12** [comment]
The font problem will be fixed in Spyder 3.2, by allowing users to set a custom scaling factor.

Please see #4454 to see how to use it.

**Comment by notuntoward** [comment]
OK, thanks.
